ops/terraform/services/eft/lambda_src/sftp_outbound_transfer/sftp_outbound_transfer.py

```diff
@@ -302,6 +303,99 @@ def _handle_s3_event(s3_object_key: str):
             send_notification(topic=partner_topic, notification=success_notif)
 
 
+def poll_queue(loop_forever):
+
+    sqs_client = boto3.client("sqs", config=BOTO_CONFIG)
+    url_response = sqs_client.get_queue_url(QueueName=EFT_OUTBOUND_QUEUE_NAME)
+    queue_url = url_response["QueueUrl"]
+    while True:
+        try:
+            response = sqs_client.receive_message(
+                QueueUrl=queue_url,
+                MaxNumberOfMessages=1,
+                WaitTimeSeconds=10,
+            )
+            messages = response.get("Messages", [])
+
+            for message in messages:
+                sqs_message = json.loads(message["Body"])
+                s3_message = json.loads(sqs_message["Message"])
+                handle_sqs_message(s3_message)
+                sqs_client.delete_message(
+                    QueueUrl=queue_url, ReceiptHandle=message["ReceiptHandle"]
+                )
+        except botocore.exceptions.ClientError as e:
+            logger.error("Error while polling SQS queue: %s", e)
+        except Exception as e:
+            logger.error("Unexpected error while polling SQS queue: %s", e)
+        finally:
+            # sleep for 5 minutes
+            if not loop_forever:
```

Comment:
Is `loop_forever` only set to false for tests? If so, do we need to sleep for 5 minutes? Would it be better to just test a method that does one loop iteration?

Reply:
I'm not sure what you're asking. The loop_forever is set in prod so it can continuously check the queue. The five minute sleep is so it doesn't continously bombard the SQS queue with requests. Yes, it's set to false for tests.

Reply:
Oh sorry, I misread part of the code there. Setting `WaitTimeSeconds` will already enable long polling so it's not continuously sending requests to SQS, but if we're expecting messages to come in very infrequently I suppose that will reduce costs somewhat.